File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/PresentationEvaluatorsFeedback.ts
import {ResearcherPresentation} from "./ResearcherPresentation";
import {ResearcherProposal} from "./ResearcherProposal";
import {NewMember} from "./NewMember";
import {ExpertEvaluator} from "../../settings/models/expert-evaluator.model";
import {PresentationReport} from "./PresentationReport";

export class PresentationEvaluatorsFeedback {

    id: number;
    uuid: string;
    m2ResearcherPresentationId: number;
    m1ResearcherProposalId: number;
    researcherUserId: number;
    stProfileOfExpertEvaluatorsId: number;
    m2AddNewMemberId: number;
    isPresent: boolean;
    evaluatorFeedback: string;
    pageNo1: number;
    researcherFeedback: string;
    pageNo2: number;
    feedbackCompleted: boolean;
    isResearcherVisible: boolean;
    isNew: boolean;
    isVisible: boolean;
    isEditable: boolean;
    type: string;

    researcherPresentationResponseDto: ResearcherPresentation;
    researcherProposalResponseDto: ResearcherProposal;
    researcherProposal: ResearcherProposal;
    newMemberResponseDto: NewMember;
    expertEvaluatorResponseDto: ExpertEvaluator;
    presentationReportResponseDto: PresentationReport;
    presentationEvaluatorsFeedbackList: PresentationEvaluatorsFeedback[];
}

File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/PresentationEvaluatorsFeedback.java
package com.ibcs.idsdp.rpm.model.domain;

import javax.persistence.*;

import com.ibcs.idsdp.common.model.domain.BaseEntity;
import com.sun.istack.NotNull;

import lombok.Data;

@Data
@Entity
@Table(name = "m2_presentation_evaluators_feedback")
public class PresentationEvaluatorsFeedback extends BaseEntity {

	@NotNull
	@ManyToOne
	@JoinColumn(name = "m2_researcher_presentation_id")
	private ResearcherPresentation researcherPresentation;
	
	@NotNull
	@ManyToOne
	@JoinColumn(name = "m1_researcher_proposal_id")
	private ResearcherProposal researcherProposal;

	private Long researcherUserId;
	private Long stProfileOfExpertEvaluatorsId;

	@ManyToOne
	@JoinColumn(name = "m2_add_new_member_id")
	private NewMember newMember;
	
	
	private Boolean isPresent;

	@Column(columnDefinition = "TEXT")
	private String evaluatorFeedback;
	private Long pageNo1;

	@Column(columnDefinition = "TEXT")
	private String researcherFeedback;

	private Long pageNo2;
	private Boolean feedbackCompleted;

	private Boolean isResearcherVisible;
	private Boolean isNew;
	private Boolean isVisible;
	private Boolean isEditable;
	
	

}
